lib/rubocop/cop/style/file_read.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Favor `File.(bin)read` convenience methods.
+      #
+      # @example text mode
+      #   # bad
+      #   File.open(filename).read
+      #   File.open(filename, &:read)
+      #   File.open(filename) { |f| f.read }
+      #   File.open(filename) do |f|
+      #     f.read
+      #   end
+      #   File.open(filename, 'r').read
+      #   File.open(filename, 'r', &:read)
+      #   File.open(filename, 'r') { |f| f.read }
+      #   File.open(filename, 'r') do |f|
+      #     f.read
+      #   end
+      #
+      #   # good
+      #   File.read(filename)
+      #
+      # @example binary mode
+      #   # bad
+      #   File.open(filename, 'rb').read
+      #   File.open(filename, 'rb', &:read)
+      #   File.open(filename, 'rb') { |f| f.read }
+      #   File.open(filename, 'rb') do |f|
+      #     f.read
+      #   end
+      #
+      #   # good
+      #   File.binread(filename)
+      #
+      class FileRead < Base
+        extend AutoCorrector
+
+        MSG = 'Use `File.%<read_method>s`.'
+
+        RESTRICT_ON_SEND = %i[open].to_set.freeze
+
+        READ_FROM_BOF_MODES = %w[r rt rb r+ r+t r+b].to_set.freeze
+
+        # @!method file_open_read(node)
+        def_node_matcher :file_open_read, <<~PATTERN
+          {
+            (block
+              (send
+                $(const {nil? cbase} :File) :open $_
+              )
+              (args _)
+              (send _ :read)
+            )
+            (send
+              (send
+                $(const {nil? cbase} :File) :open $_
+              )
+              :read
+            )
+            (send
+              $(const {nil? cbase} :File) :open $_ (block-pass (sym :read))
+            )
+          }
+        PATTERN
+
+        # @!method file_open_read_with_mode(node)
+        def_node_matcher :file_open_read_with_mode, <<~PATTERN
+          {
+            (block
+              (send
+                $(const {nil? cbase} :File) :open $_ (str $%READ_FROM_BOF_MODES)
```

Comment:
Wondering if it's possible to use `$[? %READ_FROM_BOF_MODES]` and collapse `file_open_read_with_mode` and `file_open_read` into one node pattern.

Reply:
It works with `(str $%READ_FROM_BOF_MODES)?`. Will be in the next push.